sentry-android-okhttp/src/main/java/io/sentry/android/okhttp/SentryOkHttpEvent.kt

```diff
@@ -0,0 +1,142 @@
+        breadcrumb.setData("method", method)
+        breadcrumb.setData("success", true)
+
+        // We add the same data to the root call span
+        callRootSpan?.setData("url", url)
+        callRootSpan?.setData("host", host)
+        callRootSpan?.setData("path", encodedPath)
+        callRootSpan?.setData("http.method", method)
+        callRootSpan?.setData("success", true)
+    }
+
+    /**
+     * Sets the [Response] that will be sent in the breadcrumb [Hint].
+     * Also, it sets the protocol and status code in the breadcrumb and the call root span.
+     */
+    fun setResponse(response: Response) {
+        this.response = response
+        breadcrumb.setData("protocol", response.protocol.name)
+        breadcrumb.setData("status_code", response.code)
+        callRootSpan?.setData("protocol", response.protocol.name)
+        callRootSpan?.setData("status_code", response.code)
+        callRootSpan?.status = SpanStatus.fromHttpStatusCode(response.code)
+    }
+
+    fun setProtocol(protocolName: String?) {
+        if (protocolName != null) {
+            breadcrumb.setData("protocol", protocolName)
+            callRootSpan?.setData("protocol", protocolName)
+        }
+    }
+
+    fun setRequestBodySize(byteCount: Long) {
+        if (byteCount > -1) {
+            breadcrumb.setData("request_content_length", byteCount)
+            callRootSpan?.setData("http.request_content_length", byteCount)
+        }
+    }
+
+    fun setResponseBodySize(byteCount: Long) {
+        if (byteCount > -1) {
+            breadcrumb.setData("response_content_length", byteCount)
+            callRootSpan?.setData("http.response_content_length", byteCount)
+        }
+    }
+
+    /**
+     * Sets the success flag in the breadcrumb and the call root span to false.
+     * Also sets the [errorMessage] if not null.
+     */
+    fun setError(errorMessage: String?) {
+        breadcrumb.setData("success", false)
+        callRootSpan?.setData("success", false)
+        if (errorMessage != null) {
+            breadcrumb.setData("error_message", errorMessage)
+            callRootSpan?.setData("error_message", errorMessage)
+        }
+    }
+
+    /** Starts a span, if the callRootSpan is not null. */
+    fun startSpan(event: String) {
+        // Find the parent of the span being created. E.g. secureConnect is child of connect
+        val parentSpan = when (event) {
+            // PROXY_SELECT, DNS, CONNECT and CONNECTION are not children of one another
+            SECURE_CONNECT_EVENT -> eventSpans[CONNECT_EVENT]
+            REQUEST_HEADERS_EVENT -> eventSpans[CONNECTION_EVENT]
+            REQUEST_BODY_EVENT -> eventSpans[CONNECTION_EVENT]
+            RESPONSE_HEADERS_EVENT -> eventSpans[CONNECTION_EVENT]
+            RESPONSE_BODY_EVENT -> eventSpans[CONNECTION_EVENT]
+            else -> callRootSpan
+        } ?: callRootSpan
+        val span = parentSpan?.startChild("http.client.details", event) ?: return
```

Comment:
What does `details` mean here? it's not part of the spec right.

Reply:
`http.client` is the operation of the "root" span of the http call only.
Based on [this comment](https://github.com/getsentry/sentry-java/pull/2659#discussion_r1196974000) we thought it makes sense to change the operation of the sub-spans in `http.client.details`

Reply:
It's just an assumption, though I'd rather ask if that is the case before doing the change (because in the end, they are all part of the HTTP call anyway), the spec is there to be followed, if indeed we need a new one, let's add it to the spec.
Also `details` is too generic, I'd rather go with `.dns`, `.connect`, etc, because that would make sense for pretty much every platform.

Reply:
Not an assumption, here you can see a bunch of entries for `http.client` in particular for starfish https://github.com/search?q=repo%3Agetsentry%2Fsentry%20%22http.client%22&type=code

E.g. this one doesn't account that there's something different in the description https://github.com/getsentry/sentry/blob/67c64a7a7743b384b6c5301a50ccd01b312b2aa0/static/app/components/events/interfaces/spans/utils.tsx#L288-L302

Reply:
also, checked python,php,ruby - all of them adhere to "method + url" pattern in the description, so I guess we should be cautious here. And also we should update the spec with that, if we're relying on that pattern on the product side

Reply:
I understand that Sentry itself relies on the `http.client` op, but my question is, adding more spans with that op which is an inner span of an HTTP call anyway (which is also a http.client op), why would that be a problem? that is my question?
I'd not change things just because we are assuming that we will have a problem.
And again, not against adding a suffix, I just think that `details` is too generic as well since we clearly know what is a connect, a dsn, a proxy, etc.

Reply:
the problem would be that these new `http.client` spans do not contain `method + url` in their description, as far as I found out we rely on that on the product side

Reply:
it would be ok to change the op to `http.client.dns`, `http.client.connection`, etc. 
But what would we put in the description in that case? Are we fine with having a repetition?
E.g. op = `http.client.connection`, description = `connection`

Reply:
`description` is an optional field, if there's nothing useful (specific to the action), just leave it, if there's something useful, then we put something.